addons/stock_picking_batch/models/stock_move.py

```diff
@@ -26,3 +26,11 @@ def _assign_picking_post_process(self, new=False):
         super()._assign_picking_post_process(new=new)
         for picking in self.picking_id:
             picking._find_auto_batch()
+
+    def write(self, vals):
+        res = super().write(vals)
+        if 'state' in vals and vals['state'] == 'assigned':
+            for picking in self.picking_id.filtered(lambda p: p.state == 'assigned'):
```

Comment:
put the filter in the loop to avoid doing it twice
```python
for picking in self:
    if picking.state != 'assigned':
        continue
    ...
```